repo_policy_compliance/check.py

```diff
@@ -65,6 +67,41 @@ class Report(NamedTuple):
 
 log.setup()
 
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def github_exceptions_to_fail_report(func: Callable[P, R]) -> Callable[P, R | Report]:
+    """Catch exceptions and convert to failed report with reason.
+
+    Args:
+        func: The function to catch the GithubClient exceptions for.
+
+    Returns:
+        The function where any exceptions raised would be converted to a failed result.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R | Report:
+        """Replace function.
+
+        Args:
+            args: The positional arguments passed to the original method.
+            kwargs: The keywords arguments passed to the original method.
+
+        Returns:
+            Failed result report if any exceptions were raised. The return value after calling the
+            wrapped function otherwise.
+        """
+        try:
+            return func(*args, **kwargs)
+        except GithubClientError as exc:
+            return Report(result=Result.FAIL, reason=str(exc))
```

Comment:
Not sure if we should return every error (like the example with an invalid github token in your tests) as an error reason here. I mean, it might really be an internal server error and none of the user's business to know about (I think the user will see this output in a step in their workflow). Maybe we need to distinguish more between internal server errors and user input related errors?
Thoughts, @jdkandersson  ?

Reply:
How about adding a message "Please contact the GitHub runner provider for assistance"? So the user knows what the issue is and what actions to take about it? It seems that quite a lot of users take some time trying to debug the issues themselves before reaching out to us.

Reply:
Yes, I just want to avoid sensitive data (such as the token itself) going out and being visible in github job logs.